virtcontainers/sandbox.go

```diff
@@ -944,7 +944,14 @@ func (s *Sandbox) startVM() error {
 	// Once startVM is done, we want to guarantee
 	// that the sandbox is manageable. For that we need
 	// to start the sandbox inside the VM.
-	return s.agent.startSandbox(s)
+	err := s.agent.startSandbox(s)
+	if err != nil {
+		//VM is started  but sandbox is not created successfully in VM
+		//so we need to rollback to stop VM qemu process
+		s.hypervisor.stopSandbox()
```

Comment:
Sorry I didn't realize we were talking about `hypervisor.stopSandbox()`....
Could you isolate this into its own commit as this addresses the rollback of the VM.
Also, you should follow the same model you followed for the proxy rollback, meaning you defer the `hypervisor.stopSandbox()` after you properly started the VM. This way you will both cover the error case from `waitSandbox()` and `agent.startSandbox()` that could happen.